services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes_resolver.py

```diff
@@ -82,28 +85,63 @@ def _volume_name(cls, path: Path) -> str:
         return f"{path}".replace(os.sep, "_")
 
     @classmethod
-    def source(cls, compose_namespace: str, path: Path) -> str:
-        return f"{compose_namespace}{cls._volume_name(path)}"
+    def source(cls, path: Path, node_uuid: NodeID, run_id: UUID) -> str:
+        """
+        Guarantees that the volume name is unique between runs while also
+        taking into consideration the limit for the volume name's length
+        (255 characters).
+
+        Example:
+        >>> source(
+            path="/this/is/a/data/folder",
+            node_uuid="17f5b46d-505c-489e-9818-f6d294c892d9",
+            run_id="17f5b46d-505c-489e-9818-f6d294c892d9"
+        )
+        dyv_17f5b46d-505c-489e-9818-f6d294c892d9_this_is_a_data_folder_17f5b46d-505c-489e-9818-f6d294c892d9
+        """
+        # NOTE: this name is only used when removing the volume.
+        # NOTE: issues can occur when the paths of the mounted outputs, inputs
```

Comment:
so what, this will happen right? we have no control on these 3rd party services. will there be a meaningful log when it will happen?

Reply:
As discussed I've used the inverted version here, to avoid collision issue.